### ISSUE TITLE
Canvas pattern new option to control the behavior during zoom

### ISSUE THREAD

**Issue by nobodo** [openlayers/openlayers#14443]
The current canvas pattern is nice, but the visuals get a bit 'hectic' during zooming.

Here is a mockup of how it could behave:

https://codesandbox.io/s/canvas-gradient-pattern-forked-srcd3z?file=/main.js

Finland shows the existing pattern method and other countries show the desired way. The difference is evident during zooming. Please ignore what happens during panning - for panning the existing method is ok.

So instead of constantly repositioning itself, it should have clear and concise anchor point so that the zooming would not flicker it around.

**Comment by mike-000** [comment]
The rendering is optimised for styling vector tiles.  Using a vector image layer https://codesandbox.io/s/canvas-gradient-pattern-forked-4h9izy?file=/main.js or redefining the vector layer and source as a vector tile layer with single zoom level source https://codesandbox.io/s/canvas-gradient-pattern-forked-1i5bec?file=/main.js can improve the user experience.

**Comment by nobodo** [comment]
Yes, I tried that vector image layer already and probably go with that. Still I hope someday this would get implemented.

**Comment by ahocevar** [comment]
@nobodo If you want to work on this, a new `gradientPatternOrigin` fill style option would make a nice pull request.

**Comment by nobodo** [comment]
Yes, I assume that if the pattern origin could be set to some fixed point instead of extent top left (did I understand it correctly?), then it would behave as wanted.

In ol/render/canvas/Immediate.js, setImageStyle(imageStyle) takes the origin from: 
const imageOrigin = imageStyle.getOrigin();

Is this the one that affects eventually?

**Comment by ahocevar** [comment]
No, that's for styling images. For fill patterns/gradients, the executor's fill_ method is used:

https://github.com/openlayers/openlayers/blob/f20df54d84c63d9447347284430935eeefad5663/src/ol/render/canvas/Executor.js#L558-L564

The `alignFill_` property comes from the fill instruction, and is set to true whenever a pattern or gradient is used as fill style:

https://github.com/openlayers/openlayers/blob/main/src/ol/render/canvas/Builder.js#L539-L542
